### PR TITLE
[ntuple] Provide a common set of performance counters in `RPage{Sink,Source}`

### PR DESCRIPTION
**Pull request by jalopezg-r00t**
This pull-request extends `RPageSink` and `RPageSource` to provide a common set of performance counters that can be enabled in a subclass via `EnableDefaultMetrics()`.  Afterward, the set might be extended in a subclass via a call to `fMetrics.MakeCounter<...>()`.

This not only removes boilerplate code from subclasses, but also provides a useful set of regular/computed counters.  A subclass, however, is still responsible for updating the base counters.

Alternatively, subclasses may still ignore the default set of metrics and provide their own `RNTupleMetrics` object by overriding the `GetMetrics()` member function.

Closes issue #8360.

### CODE REVIEW

**Review by mxxo — COMMENTED** [review]
> Hey Javier I think everything looks pretty good! You could also take the prefix argument out of `EnableDefaultMetrics` and pass it in the initializer list: `fMetrics("RPageSinkFile")` to match what would happen if a derived class doesn't call `EnableDefaultMetrics` but this is not a big deal.

Thanks!  It's up to Jakob to decide that, but I would prefer leaving as-is, i.e. everything is taken care of in the call to `EnableDefaultMetrics()`.

File: tree/ntuple/v7/inc/ROOT/RPageStorage.hxx [hunk 1]
```diff
@@ -186,6 +200,17 @@ protected:
    static RSealedPage SealPage(const RPage &page, const RColumnElementBase &element,
       int compressionSetting, void *buf);
 
+   /// Enables the default set of metrics provided by RPageSink. `prefix` will be used as the prefix for
+   /// the counters registered in the internal RNTupleMetrics object.
+   /// Afterward, this set of counters can be extended by a subclass by calling `fMetrics.MakeCounter<...>()`.
```
Comment (jblomer) [inline tree/ntuple/v7/inc/ROOT/RPageStorage.hxx:205]:
```suggestion
   /// This set of counters can be extended by a subclass by calling `fMetrics.MakeCounter<...>()`.
```

File: tree/ntuple/v7/inc/ROOT/RPageStorage.hxx [hunk 2]
```diff
@@ -261,6 +313,14 @@ protected:
    /// Usage of this method requires construction of fDecompressor.
    std::unique_ptr<unsigned char []> UnsealPage(const RSealedPage &sealedPage, const RColumnElementBase &element);
 
+   /// Enables the default set of metrics provided by RPageSource. `prefix` will be used as the prefix for
+   /// the counters registered in the internal RNTupleMetrics object.
+   /// A subclass using the default set of metrics is always responsible for updating the counters
```
Comment (jblomer) [inline tree/ntuple/v7/inc/ROOT/RPageStorage.hxx:318]:
```suggestion
   /// A subclass using the default set of metrics is responsible for updating the counters
```

File: tree/ntuple/v7/inc/ROOT/RPageStorage.hxx [hunk 3]
```diff
@@ -313,6 +373,9 @@ public:
    /// actual implementation will only run if a task scheduler is set. In practice, a task scheduler is set
    /// if implicit multi-threading is turned on.
    void UnzipCluster(RCluster *cluster);
+
+   /// Returns the default metrics object.  Subclasses might alternatively provide their own metrics object by overriding this.
```
Comment (jblomer) [inline tree/ntuple/v7/inc/ROOT/RPageStorage.hxx:377]:
```suggestion
   /// Returns the default metrics object.  Subclasses might alternatively override the method and provide their own metrics object.
```

**Review by jblomer — CHANGES_REQUESTED** [review]
Great, many thanks! Just two tiny suggestions for the comments.

**Review by jblomer — APPROVED** [review]

**Pull request merged**